kayenta-atlas/src/main/java/com/netflix/kayenta/atlas/metrics/AtlasMetricsService.java

```diff
@@ -81,11 +81,34 @@ public boolean servicesAccount(String accountName) {
     return accountNames.contains(accountName);
   }
 
+
+  @Override
+  public List<String> getLocations(String accountName) {
+    AtlasNamedAccountCredentials credentials = (AtlasNamedAccountCredentials)accountCredentialsRepository
+      .getOne(accountName)
+      .orElseThrow(() -> new IllegalArgumentException("Unable to resolve account " + accountName + "."));
+
+    return credentials.getBackendUpdater().getBackendDatabase().getLocations();
+  }
+
+  @Override
+  public List<String> getRecommendedLocations(String accountName) {
+    AtlasNamedAccountCredentials credentials = (AtlasNamedAccountCredentials)accountCredentialsRepository
+      .getOne(accountName)
+      .orElseThrow(() -> new IllegalArgumentException("Unable to resolve account " + accountName + "."));
+
+    if (credentials.getRecommendedLocations() == null) {
```

Comment:
The if/else here are in the opposite order of the way the logic was explained in the commit comment. I think it reads more naturally as described in the comment (if it's present, use it, otherwise fall back to using `locations` instead).

Also, this will not fall back to `locations` if `recommendedLocations` is provided but is an empty list. I think this is what you intended, but wanted to check just in case.

Reply:
I'm starting to think the UI should do the fallback, and I should just return the literal lists.  This would make our code easier, and the UI can tell the difference should it care.

Reply:
Seems reasonable to me.

Reply:
I modified this to return the empty list if there are no recommendations, and let them run independently, with the UI sorting it out.